scipy/sparse/linalg/tests/test_pydata_sparse.py

```diff
@@ -0,0 +1,226 @@
+import pytest
+
+import numpy as np
+import scipy.sparse as sp
+import scipy.sparse.linalg as splin
+
+from numpy.testing import assert_allclose
+
+try:
+    import sparse
+except ImportError:
+    sparse = None
+
+pytestmark = pytest.mark.skipif(sparse is None,
+                                reason="pydata/sparse not installed")
+
+
+def test_isolve_gmres():
+    # Several of the iterative solvers use the same
+    # isolve.utils.make_system wrapper code, so test just one of them.
+    np.random.seed(1234)
+
+    A = sparse.COO(np.random.rand(5, 5))
```

Comment:
Would be good to parametrize this over `{DOK, COO, CSD}`  and `xfail` appropriately. It'll expose a lot of bugs/missing features and allow testing.

Reply:
Is this useful at this stage?

sparse.DOK does not have matvec so nothing will work, and the version currently on pypi does not have CSD.

Reply:
It should hopefully convert to COO when  there’s nothing else available.

Reply:
Rewrote the test in any case.

Reply:
It’s useful to be able to test upstream in PyData/Sparse and add features related to `xfail`.

Reply:
Well, marked them with xfail.

DOK does fail with e.g. "'DOK' object has no attribute 'to_scipy_sparse'" and similarly for attribute 'dot'

Reply:
@hameerabbasi do you mean 'GCXS' or is there a reason you're writing 'CSD'?

Reply:
I do mean GCXS... Effects of being tired and sleepless.

Reply:
I think I'll limit this PR to `{COO,DOK}`, which are what is currently available in the pypi version.